### PR TITLE
COMMERCE-2231 Refactor subscription info tag

### PR DESCRIPTION
**Pull request by alessiorendina**
Based on #1503 @marco-leo

### CODE REVIEW

File: commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/ProductSubscriptionInfoTag.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+		_subscriptionPeriodKey = null;
+	}
+
+	protected CPSubscriptionTypeRegistry cpSubscriptionTypeRegistry;
+
+	private String _getPeriodKey(String period, boolean plural) {
+		if (plural) {
+			return LanguageUtil.get(
+				request,
+				StringUtil.toLowerCase(period) + CharPool.LOWER_CASE_S);
```
Comment (ethanbustad) [inline commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/ProductSubscriptionInfoTag.java]: `TextFormatter.formatPlural` would be a more robust way of doing this.

File: commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/ProductSubscriptionInfoTag.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+		}
+
+		return period;
```
Comment (ethanbustad) [inline commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/ProductSubscriptionInfoTag.java]: Why do we localize the `period` if `plural` is true, but we don't localize if `plural` is false? Seems like we should do it both times or neither time.

**Review by ethanbustad — CHANGES_REQUESTED** [review]

**Comment by alecsloan** [comment]
ci:test

File: commerce-order-web/src/main/resources/META-INF/resources/email/detail.jsp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+			style="success"
+		/>
+	</div>
+
+	<div class="commerce-notification-queue-entry-subject">
+		<h3><%= commerceNotificationQueueEntry.getSubject() %></h3>
+	</div>
+
+	<div class="commerce-notification-queue-entry-body">
+		<%= commerceNotificationQueueEntry.getBody() %>
```
Comment (ethanbustad) [inline commerce-order-web/src/main/resources/META-INF/resources/email/detail.jsp:40]: Does the content of `getSubject` and `getBody` come from trusted sources? If not, this opens up XSS vulnerabilities.

Comment (marco-leo) [reply commerce-order-web/src/main/resources/META-INF/resources/email/detail.jsp:40]: We generate the content from when sending the notification from the notification template that is an editor

**Review by ethanbustad — APPROVED** [review]

**Pull request closed without merging**